nncf/torch/external_hook.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2023 Intel Corporation
```

Comment:
A hook is an entity that belongs to a tracing context first, perhaps this file should have been located somewhere closer to `nncf/torch/dynamic_graph`